jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/FileBufferedResponseHandler.java

```diff
@@ -229,5 +230,23 @@ protected void onFailure(Throwable cause)
             };
             icb.iterate();
         }
+
+        private ByteBuffer readByteBuffer(Path path, long pos, long len) throws IOException
+        {
+            ByteBuffer buffer = BufferUtil.toMappedBuffer(path, pos, len);
+            if (buffer != null)
+                return buffer;
+
+            // if we reached here, then file mapped byte buffers is not supported.
+            // we fall back to using traditional I/O instead.
+            try (SeekableByteChannel channel = Files.newByteChannel(path))
+            {
+                channel.position(pos);
+                buffer = ByteBuffer.allocate((int)len);
+                channel.read(buffer);
```

Comment:
IIRC `channel.read()` does not guarantee that the buffer will be filled as this API is non-blocking. You are going to either spin loop or use some form of blocking API instead.

Reply:
I don't see that as a problem, we read what we can and return, even if the buffer isn't full, the iterating calback that calls this method will just get a new position to read from, right?

Reply:
The new `FileBufferedResponseHandler.setUseFileMapping(boolean)` allows me to test these new features in unit tests too.

Reply:
No longer relevant.